interface/reports/front_receipts_report.php

```diff
@@ -185,15 +228,25 @@ function show_receipt(pid,timestamp) {
     "MAX(r.user) AS user, " .
     "p.fname, p.mname, p.lname, p.pubpid " .
     "FROM payments AS r " .
+    "JOIN form_encounter AS fe ON fe.encounter=r.encounter " .
     "LEFT OUTER JOIN patient_data AS p ON " .
     "p.pid = r.pid " .
     "WHERE " .
     "r.dtime >= ? AND " .
-    "r.dtime <= ? " .
-    "GROUP BY r.dtime, r.pid ORDER BY r.dtime, r.pid";
-
-    // echo "<!-- $query -->\n"; // debugging
-    $res = sqlStatement($query, array($from_date.' 00:00:00', $to_date.' 23:59:59'));
+    "r.dtime <= ? AND ";
+    if($_POST['form_facility']!="")
+    $query.="fe.facility_id = ? AND ";
+    if($_POST['form_provider']!="")
+    $query.="fe.provider_id = ? AND ";
+    $query.="1 GROUP BY r.dtime, r.pid ORDER BY r.dtime, r.pid";
+
+    // echo " $query \n"; // debugging
+    $inputArray=array($from_date.' 00:00:00', $to_date.' 23:59:59');
+    if($_POST['form_facility']!="")
+    $inputArray[]=$_POST['form_facility'];
+    if($_POST['form_provider']!="")
+    $inputArray[]=$_POST['form_provider'];
+    $res = sqlStatement($query, $inputArray);
```

Comment:
the following will be more clear to follow:
```
    "r.dtime <= ? AND ";
    $inputArray = [$from_date.' 00:00:00', $to_date.' 23:59:59'];

    if ($_POST['form_facility'] != "") {
        $query .= "fe.facility_id = ? AND ";
        $inputArray[]=$_POST['form_facility'];
    }

    if ($_POST['form_provider'] != "") {
        $query. = "fe.provider_id = ? AND ";
        $inputArray[] = $_POST['form_provider'];
    }

    $query .= "1 GROUP BY r.dtime, r.pid ORDER BY r.dtime, r.pid";

    $res = sqlStatement($query, $inputArray);
```

Reply:
Great Standardization guide brady. Will update and push the changes as soon as possible.